distributed/shuffle/_arrow.py

```diff
@@ -54,7 +54,9 @@ def convert_partition(data: bytes) -> pa.Table:
     while file.tell() < end:
         sr = pa.RecordBatchStreamReader(file)
         shards.append(sr.read_all())
-    return pa.concat_tables(shards)
+    table = pa.concat_tables(shards)
+    df = table.to_pandas(self_destruct=True)
+    return df.astype(meta.dtypes)
```

Comment:
This might not be the most performant version, but I also don't know if it's much of a problem. I'll run an A/B test on the existing test suite.

Reply:
My guess is it'd be good to avoid a cast is possible (at least for strings) via `type_mapper=` in `pa.Table.to_pandas()`. For example, as is, this will create object-backed `string[python]` columns first and then cast them to `string[pyarrow]`. 

This definitely isn't a blocker for this PR, but let's add a `# TODO: ` comment if we don't include that logic now

Reply:
I'll create a follow-up ticket once this is merged.

Reply:
Yes this slows stuff down significantly, mapping the strings (pa.string and pa.large_string) to pd.ArrowDtype makes this zero copy and should speed up follow up as types